aten/src/ATen/native/mps/operations/Copy.mm

```diff
@@ -335,16 +370,24 @@ void copy_blit_mps(void* dst, const void* src, size_t size) {
     needs_broadcasting = true;
   }
 
+  // copying scalars from MPS to MPS is possible with try_copy_scalars_mps(), but it
+  // may require syncing with CPU. So it's better to use blit encoder without syncing.
+  if (src.device().type() == at::kMPS && dst.device().type() == at::kMPS) {
```

Comment:
It will still be better to use Shared events here along with memcpy than using Blits. I think there is room for improvement.

Reply:
Because we need to read from an MPS source buffer, we'll need to sync before memcpy.